Api/Connection.php

```diff
@@ -21,7 +21,14 @@ public function post($url, $bodyArray = false)
                 'Connection: Close',
                 'Host: '.$parsedUrl['host']
             ];
-            return $this->curlCall($url, $headers, $encodedBody);
+            
+            if ($activity) {
+                $headers[] = 'HTTP_X_DIGEST: ' . $encodedBody;
+                
+                return $this->curlCall($url, $headers, '');
```

Comment:
You are not sending the body, why?

Reply:
Because for activity call the $signature is actually the $encodedBody data and since it's information is already present in the 'HTTP_X_DIGEST' header I thought it is unnecessary to send it again as $body.

Reply:
Did you read anything about how signatures in requests work?